nameko/timer.py

```diff
@@ -49,26 +52,29 @@ def kill(self):
 
     def _run(self):
         """ Runs the interval loop. """
-
-        sleep_time = self.interval
+        sleep_time = 0 if self.eager else self.interval
 
         while True:
             # sleep for `sleep_time`, unless `should_stop` fires, in which
             # case we leave the while loop and stop entirely
-            with Timeout(sleep_time, exception=False):
```

Comment:
@davidszotten is correct. The thread will probably spend most of its time asleep, and we have to be able to wake it when `stop` is called. We should have a test for this too, something like:

``` python
def test_stop_while_sleeping(container_factory):

    class Service(object):
        name = "service"

        @timer(interval=5)
        def tick(self):
            pass

    container = container_factory(Service, {})
    container.start()

    # raise a Timeout if the container fails to stop within 1 second
    with eventlet.Timeout(1):
        container.stop()
```